src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java

```diff
@@ -53,6 +57,44 @@ public static List checkVirtualTracks(IMFCompositionPlaylistType compositionPlay
             else if(virtualTrack.getSequenceTypeEnum().equals(Composition.SequenceTypeEnum.MainAudioSequence)){
                 foundMainAudioEssence = true;
             }
+
+            if((virtualTrack.getSequenceTypeEnum().equals(Composition.SequenceTypeEnum.MainImageSequence)
+                    || virtualTrack.getSequenceTypeEnum().equals(Composition.SequenceTypeEnum.MainAudioSequence))
+                    && compositionPlaylistType.getEssenceDescriptorList() != null
+                    && compositionPlaylistType.getEssenceDescriptorList().size() > 0)
+            {
+                //Construct a DOMNodeObjectModel object for every EssenceDescriptor that is a part of the EssenceDescriptorList
+                Map<UUID, List<DOMNodeObjectModel>> domNodeObjectModelMap = new HashMap<>();
+                List<IMFEssenceDescriptorBaseType> essenceDescriptors = compositionPlaylistType.getEssenceDescriptorList();
+                for(IMFEssenceDescriptorBaseType essenceDescriptorBaseType : essenceDescriptors){
+                    List<DOMNodeObjectModel> domNodeObjectModels = new ArrayList<>();
+                    for (Object object : essenceDescriptorBaseType.getAny()) {
+                        domNodeObjectModels.add(new DOMNodeObjectModel((Node) object));
+                    }
+                    domNodeObjectModelMap.put(essenceDescriptorBaseType.getId(), domNodeObjectModels);
+                }
+
+                List<DOMNodeObjectModel> virtualTrackEssenceDescriptors = new ArrayList<>();
+                for(IMFBaseResourceType imfBaseResourceType : virtualTrackResourceList){
+                    IMFTrackFileResourceType imfTrackFileResourceType = IMFTrackFileResourceType.class.cast(imfBaseResourceType);
+                    List<DOMNodeObjectModel> domNodeObjectModels = domNodeObjectModelMap.get(UUIDHelper.fromUUIDAsURNStringToUUID(imfTrackFileResourceType.getSourceEncoding()));
+                    virtualTrackEssenceDescriptors.addAll(domNodeObjectModels);
+                }
+
+                if(!(virtualTrackEssenceDescriptors.size() > 0)){
+                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CPL_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.FATAL, String.format("This Composition represented by the ID %s is invalid since the VirtualTrack represented by ID %s doesn't seem to refer to a single EssenceDescriptor in the CPL's EssenceDescriptorList", compositionPlaylistType.getId().toString(), virtualTrack.getTrackID().toString()));
+                }
+                boolean isVirtualTrackHomogeneous = true;
+                DOMNodeObjectModel refDOMNodeObjectModel = virtualTrackEssenceDescriptors.get(0);
+                for(int i=1; i<virtualTrackEssenceDescriptors.size(); i++){
+                    isVirtualTrackHomogeneous &= refDOMNodeObjectModel.equivalent(virtualTrackEssenceDescriptors.get(i));
+                    refDOMNodeObjectModel = virtualTrackEssenceDescriptors.get(i);
```

Comment:
Yes, we needn't however by doing this if we need to debug this code we will get to know if the first EssenceDescriptor is the odd one or not.